File: Práctica4/java/src/deepspace/WeaponType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author daniel
 */

// Enumerado que representa los tipos de armas del juego, con su
// poder asociado

public enum WeaponType {
    LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f);
    
    private final float power;
    
    WeaponType(float p){
        this.power = p;
    }
    
    float getPower(){
        return power;
    }   
}

File: Práctica4/Práctica 4-20210523/lib/WeaponType.rb
#encoding:utf-8

module Deepspace
  module WeaponType

    class Type

      def initialize(pow, _name)
        @POWER = pow
        @name = _name
      end

      def power
        return @POWER
      end

      def to_s
        return "#{@name}"
      end
    end

    LASER = Type.new(2.0, "LASER")
    MISSILE = Type.new(3.0, "MISSILE")
    PLASMA = Type.new(4.0, "PLASMA")
  end
end
    
